File: app/src/main/java/com/etologic/pointscorer/MyAnimationUtils.java
package com.etologic.pointscorer;

import android.view.animation.AlphaAnimation;
import android.view.animation.Animation;
import android.view.animation.AnimationSet;
import android.view.animation.DecelerateInterpolator;
import android.view.animation.ScaleAnimation;
import android.widget.TextView;

import java.util.Locale;

import static android.view.animation.Animation.RELATIVE_TO_SELF;

public class MyAnimationUtils {

    public static Animation getUpdatePointsAnimation(TextView tvPoints, TextView tvPointsForAnimation, int points) {
        AlphaAnimation alphaAnimation = new AlphaAnimation(0, 1);
        alphaAnimation.setDuration(1000);
        ScaleAnimation scaleAnimation = new ScaleAnimation(4, 1, 4, 1, RELATIVE_TO_SELF, .5f, RELATIVE_TO_SELF, .3f);
        alphaAnimation.setDuration(1000);
        AnimationSet animationSet = new AnimationSet(true);
        animationSet.addAnimation(alphaAnimation);
        animationSet.addAnimation(scaleAnimation);
        animationSet.setDuration(500);
        animationSet.setInterpolator(new DecelerateInterpolator());
        animationSet.setAnimationListener(new Animation.AnimationListener() {
            @Override public void onAnimationStart(Animation animation) {
                tvPointsForAnimation.setText(String.format(Locale.getDefault(), "%d", points));
            }
            @Override public void onAnimationEnd(Animation animation) {
                tvPoints.setText(String.format(Locale.getDefault(), "%d", points));
            }
            @Override public void onAnimationRepeat(Animation animation) {}
        });
        return animationSet;
    }
}

File: app/src/main/java/com/etologic/pointscorer/app/common/utils/MyAnimationUtils.kt
package com.etologic.pointscorer.app.common.utils

import android.view.animation.AlphaAnimation
import android.view.animation.Animation
import android.view.animation.Animation.AnimationListener
import com.etologic.pointscorer.common.Constants.A_SECOND_IN_MILLIS

object MyAnimationUtils {

    fun getAuxPointsFadeOutAnimation(endAction: (() -> Unit)?): Animation {
        val alphaAnimation = AlphaAnimation(1f, 0f)
        alphaAnimation.duration = 5 * A_SECOND_IN_MILLIS
        alphaAnimation.fillAfter = true
        alphaAnimation.setAnimationListener(object : AnimationListener {
            override fun onAnimationStart(animation: Animation) {}
            override fun onAnimationEnd(animation: Animation) {
                endAction?.invoke()
            }
            override fun onAnimationRepeat(animation: Animation) {}
        })
        return alphaAnimation
    }

}
